frontend/src/redux/Room.ts

```diff
@@ -0,0 +1,32 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getRoom, Room } from '../api/Room';
+
+const initialState = null as Room | null;
+
+// Create an async action that fetches the room from the backend
+export const fetchRoom = createAsyncThunk<Room, string>(
+  'rooms/fetch',
+  async (roomId, thunkApi) => getRoom(roomId)
+    .then((res) => res)
+    .catch((err) => thunkApi.rejectWithValue(err)),
+);
```

Comment:
I guess mainly it was for convenience - since fetchRoom is used a lot, having this method could probably save on a bit of code from something like this:

```tsx
getRoom(roomId)
  .then((res) => dispatch(setRoom(res));
```

to this:

```tsx
dispatch(fetchRoom(roomId))
```

Though maybe in the future when we have more complex async actions then these thunks might become more noticeably useful